Musicca/app/src/main/java/com/example/musicca/activities/SongPlaylistActivity.java

```diff
@@ -0,0 +1,100 @@
+package com.example.musicca.activities;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.example.musicca.R;
+import com.example.musicca.models.Playlist;
+import com.example.musicca.models.Song;
+import com.parse.GetCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+
+public class SongPlaylistActivity extends AppCompatActivity {
+
+    Playlist currentPlaylist;
+    Song currentSong;
+
+    private ImageView ivSongAlbum;
+    private TextView tvTitle;
+    private TextView tvArtist;
+    private ImageView ivPrevious;
+    private ImageView ivPlayPause;
+    private ImageView ivNext;
+    private Button btnReturnPlaylist;
+
+    private String albumUrl;
+    private String playlistObjectId;
+    private String songObjectId;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_song_playlist);
+
+        ivSongAlbum = findViewById(R.id.ivSongAlbum);
+        tvTitle = findViewById(R.id.tvTitle);
+        tvArtist = findViewById(R.id.tvArtist);
+        btnReturnPlaylist = findViewById(R.id.btngotoPlaylist);
+
+        songObjectId = getIntent().getStringExtra("songObjectid");
+        playlistObjectId = getIntent().getStringExtra("playlistobjectid");
+
+        albumUrl = getIntent().getStringExtra("albumiconurl");
+        Glide.with(this).load(albumUrl).into(ivSongAlbum);
+
+        tvTitle.setText(getIntent().getStringExtra("songtitle"));
+        tvArtist.setText(getIntent().getStringExtra("songartist"));
+
+        currentPlaylist = getCurrentPlaylist(playlistObjectId);
+        currentSong = getCurrentSong(songObjectId);
+
+        btnReturnPlaylist.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                gotoPlaylist();
+            }
+        });
+    }
+
+    private Playlist getCurrentPlaylist(String playlistobjectid) {
+        final Playlist[] currentplaylist = new Playlist[1];
```

Comment:
If you make `currentplaylist` a member variable of the class, you won't need to make it an array.